lib/Data/Validate/Sanctions/Fetcher.pm

```diff
@@ -528,4 +542,65 @@ sub _entries_from_remote_src {
     return $entries // die "An error occurred while fetching data from '$src_url' due to $error_log\n";
 }
 
+=head2 _epoch_to_date
+
+Converts an epoch timestamp to a date string in YYYY-MM-DD format.
+
+  my $date = $fetcher->_epoch_to_date(1672444800);
+
+=cut
+
+sub _epoch_to_date {
+    my $epoch = shift;
+
+    # Input validation
+    die "Epoch timestamp must be defined" unless defined $epoch;
+
+    # Convert epoch to DateTime object
+    my $dt = DateTime->from_epoch(epoch => $epoch);
+
+    # Return formatted date string
+    return $dt->ymd('-');
+}
+
+=head2 _clean_url
+
+Removes specific query parameters from a URL.
+
+  my $clean_url = $fetcher->_clean_url($url);
+
+=cut
+
+sub _clean_url {
+    my $url = shift;
+
+    # Remove the token parameter from the URL
+    $url =~ s/[?&]token=[^&]+//;
+
+    return $url;
+}
+
+=head2 _create_hash
+
+Converts the data to a string and creates a SHA-256 hash.
+
+  my $hash = $fetcher->create_hash($data);
+
+=cut
+
+sub _create_hash {
+    my ($data) = @_;
+
+    # Convert the data to a JSON string
+    my $json_string = to_json(
+        $data,
+        {
+            canonical => 1,
+            utf8      => 1
+        });
+
+    # Generate and return the SHA-256 hash
+    return sha256_hex($json_string);
+}
+
```

Comment:
Nicely done. Should have done something similar for my stuff as we are doing similar computing in almost all the xml parsing.